app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/BaseActivity.java

```diff
@@ -8,10 +9,26 @@
 import android.support.v7.app.AppCompatActivity;
 import android.view.MenuItem;
 
+import nerd.tuxmobil.fahrplan.congress.net.ConnectivityStateReceiver;
 import nerd.tuxmobil.fahrplan.congress.utils.ActivityHelper;
 
 public abstract class BaseActivity extends AppCompatActivity {
 
+    private ConnectivityStateReceiver connectivityStateReceiver;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
Do you need to listen to network change events while the app is in the background? Wouldn't registering in `onResume` and unregistering in `onPause` be enough? Once you target Android 8 you won't be able to start a background service if your app is not in the foreground.

https://developer.android.com/about/versions/oreo/background#services

Reply:
For now (targetSdkVersion 24) the app receives network change events when the app is in background. I consider replacing the mechanism with something such as WorkManager but I did not have the time to look into this. My current focus is to raise the targetSdkVersion stepwise - see issues.